prow/cmd/deck/rerun_test.go

```diff
@@ -232,6 +235,512 @@ func TestRerun(t *testing.T) {
+							GitHubUsers:   []string{"authorized", "alsoauthorized"},
+							GitHubTeamIDs: []int{42},
+						},
+					},
+					Brancher: config.Brancher{
+						Branches: []string{"master"},
+					},
+				}}
+			cfg := func() *config.Config {
+				return &config.Config{
+					JobConfig: config.JobConfig{
+						PresubmitsStatic: map[string][]config.Presubmit{"org/repo": jobs},
+					},
+				}
+			}
+			if err := config.SetPresubmitRegexes(jobs); err != nil {
+				t.Fatal(err)
+			}
+			handler := handleRerun(cfg, fakeProwJobClient.ProwV1().ProwJobs("prowjobs"), tc.rerunCreatesJob, authCfgGetter, goa, ghc, rc, &pca, logrus.WithField("handler", "/rerun"))
+			handler.ServeHTTP(rr, req)
+			if rr.Code != tc.httpCode {
+				t.Fatalf("Bad error code: %d", rr.Code)
+			}
+
+			if tc.shouldCreateProwJob {
+				pjs, err := fakeProwJobClient.ProwV1().ProwJobs("prowjobs").List(context.Background(), metav1.ListOptions{})
+				if err != nil {
+					t.Fatalf("failed to list prowjobs: %v", err)
+				}
+				if numPJs := len(pjs.Items); numPJs != 2 {
+					t.Errorf("expected to get two prowjobs, got %d", numPJs)
+				}
+
+			} else if !tc.rerunCreatesJob && tc.httpCode == http.StatusOK {
+				resp := rr.Result()
+				defer resp.Body.Close()
+				body, err := ioutil.ReadAll(resp.Body)
+				if err != nil {
+					t.Fatalf("Error reading response body: %v", err)
+				}
+				var res prowapi.ProwJob
+				if err := yaml.Unmarshal(body, &res); err != nil {
+					t.Fatalf("Error unmarshaling: %v", err)
+				}
+				if res.Spec.Job != "whoa" {
+					t.Errorf("Wrong job, expected \"whoa\", got \"%s\"", res.Spec.Job)
+				}
+				// Label/annotation from new configuration
+				if val, ok := res.Labels["foo"]; !ok {
+					t.Errorf("Wrong label, expected \"foo\", got \"%s\"", val)
+				}
+				if val, ok := res.Annotations["foo"]; !ok {
+					t.Errorf("Wrong label, expected \"foo\", got \"%s\"", val)
+				}
+				// Test that component generated label/annotation are copied over
+				if val, ok := res.Labels[client.GerritReportLabel]; !ok {
+					t.Errorf("Wrong label, expected \"prow.k8s.io/gerrit-report-label\", got \"%s\"", val)
+				}
+				if val, ok := res.Annotations[client.GerritID]; !ok {
+					t.Errorf("Wrong annotation, expected \"prow.k8s.io/gerrit-id\", got \"%s\"", val)
+				}
+				// Should not copy over labels/annotations we don't care about
+				if val, ok := res.Labels["random"]; ok {
+					t.Errorf("Wrong label, should not expect \"random\", got \"%s\"", val)
+				}
+				if val, ok := res.Annotations["random"]; ok {
+					t.Errorf("Wrong annotation, should not expect \"random\", got \"%s\"", val)
+				}
+				if res.Status.State != prowapi.TriggeredState {
+					t.Errorf("Wrong state, expected \"%v\", got \"%v\"", prowapi.TriggeredState, res.Status.State)
+				}
+			}
+		})
+	}
+}
+
+// TestLatestRerunPeriodic just checks that the result can be unmarshaled properly, has an
+// updated status, and has equal spec.
+func TestLatestRerunPeriodic(t *testing.T) {
```

Comment:
From a quick glance this looks very similiar to TestLatestRerun, the only difference is the Prowjob that is being used - Would it make sense to deduplicate the two and make the prowjob a property of the testcase?

Reply:
Updated!